ecommerce/src/main/kotlin/io/hhplus/ecommerce/product/application/usecase/ProductRankingQueryUseCase.kt

```
@@ -0,0 +1,167 @@
+            salesCount = salesCount
+        )
+    }
+
+    /**
+     * 랭킹 데이터를 응답 DTO로 변환
+     */
+    private fun toRankingResponses(topProducts: List<Pair<Long, Long>>): List<ProductRankingResponse> {
+        return topProducts.mapIndexed { index, (productId, salesCount) ->
+            val product = try {
+                productDomainService.getProduct(productId)
+            } catch (e: Exception) {
```

detekt (1.23.8)

[warning] The caught exception is too generic. Prefer catching specific exceptions to the case that is currently handled.

(detekt.exceptions.TooGenericExceptionCaught)